### PR TITLE
feat: add syncs API in compass

### PR DESCRIPTION
**Pull request by batrov**
- Add schema for compass's sync API

Issue: https://github.com/goto/compass/issues/67

### CODE REVIEW

File: gotocompany/compass/v1beta1/service.proto [hunk 1]
```diff
@@ -403,6 +403,20 @@ service CompassService {
     };
   }
 
+  rpc SyncAssets(SyncAssetsRequest) returns (SyncAssetsResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/assets/sync"
```
Comment (haveiss) [inline gotocompany/compass/v1beta1/service.proto:408]: let's don't use `get` since we are mutating data here

File: gotocompany/compass/v1beta1/service.proto [hunk 2]
```diff
@@ -1198,6 +1212,12 @@ message CreateAssetProbeResponse {
   string id = 1;
 }
 
+message SyncAssetsRequest {}
+
+message SyncAssetsResponse {
+  uint32 num_affected = 1;
```
Comment (haveiss) [inline gotocompany/compass/v1beta1/service.proto:1218]: are we planning to have immediate result when calling the API? it the data is huge, it will take a long time and lead to timeout

**Review by haveiss — CHANGES_REQUESTED** [review]

File: gotocompany/compass/v1beta1/service.proto [hunk 3]
```diff
@@ -1198,6 +1212,12 @@ message CreateAssetProbeResponse {
   string id = 1;
 }
 
+message SyncAssetsRequest {}
```
[the remark this answers is not in the document]
Comment (batrov) [reply gotocompany/compass/v1beta1/service.proto]: yea make sense, updated to also accept index_names to target specific index to be recreated

File: gotocompany/compass/v1beta1/service.proto [hunk 4]
```diff
@@ -1198,6 +1212,14 @@ message CreateAssetProbeResponse {
   string id = 1;
 }
 
+message SyncAssetsRequest {
+  repeated string index_names = 1;
+}
+
+message SyncAssetsResponse {
+  string job_id = 1;
```
Comment (bsushmith) [inline gotocompany/compass/v1beta1/service.proto:1220]: do you also want to add a job status check api ? will it be in a separate PR ? @batrov

Comment (batrov) [reply gotocompany/compass/v1beta1/service.proto:1220]: I think we can consider that later, for now we will only throw status accepted after triggering this API

File: gotocompany/compass/v1beta1/service.proto [hunk 5]
```diff
@@ -403,6 +403,20 @@ service CompassService {
     };
   }
 
+  rpc SyncAssets(SyncAssetsRequest) returns (SyncAssetsResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/assets/sync"
```
Comment (haveiss) [inline gotocompany/compass/v1beta1/service.proto:408]:
```suggestion
      post: "/v1beta1/assets/sync"
      body: "*"
```

File: gotocompany/compass/v1beta1/service.proto [hunk 6]
```diff
@@ -1198,6 +1212,12 @@ message CreateAssetProbeResponse {
   string id = 1;
 }
 
+message SyncAssetsRequest {
+  repeated string index_names = 1;
```
Comment (haveiss) [inline gotocompany/compass/v1beta1/service.proto:1216]: Let's use this instead?
https://github.com/goto/proton/blob/34aa07ae5769b44043beb3e98da7d927904ac557/gotocompany/compass/v1beta1/service.proto#L974-L985

Comment (batrov) [reply gotocompany/compass/v1beta1/service.proto]: updated, considering to filter by services for simplifying things

**Review by haveiss — APPROVED** [review]

**Review by bsushmith — APPROVED** [review]
